fast64_internal/oot/oot_utility.py

```diff
@@ -599,3 +630,260 @@ def oot_utility_register():
 def oot_utility_unregister():
 	for cls in reversed(oot_utility_classes):
 		unregister_class(cls)
+
+def getActorParameter(object, field, shift, mask):
+	'''Returns the actor parameter with the shifts'''
+	attr = getattr(object, field, '0x00')
+	if attr != '0x00' and attr is not None:					
+		if isinstance(attr, str):
+			if shift != '0':
+				return f"(({attr} << {shift}) & {mask}) | "
+			else:
+				return f"({attr} & {mask}) | " 
+		elif isinstance(attr, bool) and attr:
+			return f"(1 << {shift}) | "
+		else:
+			return ""
+	else:
+		return ""
+
+def getActorFinalParameters(detailedProp, actorKey, paramTarget, field):
+	'''Returns the full parameter string'''
+	params = ""
+	if field != None:
+		panelParams = getattr(detailedProp, field, "")
+		if panelParams == "": return "0x0"
+	for actorNode in root:
+		if actorKey == actorNode.get('Key'):
+			lenProp = getMaxElemIndex(actorKey, 'Property', None)
+			lenSwitch = getMaxElemIndex(actorKey, 'Flag', 'Switch')
+			lenBool = getMaxElemIndex(actorKey, 'Bool', None)
+			lenEnum = getMaxElemIndex(actorKey, 'Enum', None)
+			for elem in actorNode:
+				actorType = getattr(detailedProp, actorKey + '.type', None)
+				if hasTiedParams(elem.get('TiedParam'), actorType):
+					params += getActorString(elem, detailedProp, actorKey, lenProp, lenSwitch, lenBool, lenEnum, paramTarget)
+	params = params.rstrip(' | ')
```

Comment:
```suggestion
	params = params.rstrip().rstrip('|').rstrip()
```
may be more what you intended here. this handles any whitespace (rstrip() with no arguments), not just spaces